rack-protection/rack-protection.gemspec

```diff
@@ -17,6 +17,11 @@ Gem::Specification.new do |s|
     "Gemfile",
     "rack-protection.gemspec"
   ]
+  s.metadata = {
+    'source_code_uri' => 'https://github.com/sinatra/sinatra/tree/master/rack-protection'
+    'homepage_uri'      => 'http://sinatrarb.com/protection/',
+    'documentation_uri' => 'https://www.rubydoc.info/gems/rack-protection'
+  }
```

Comment:
I'd like to raise an error with kindness message `unless s.respond_to?(:metadata)`. Should be something like:

```ruby
if s.respond_to?(:metadata)
  s.metadata = {}
else
  raise 'RubyGems 2.0 or newer is required to protect against public gem pushes.'
end
```